File: programmers/DFS&BFS/네트워크.cpp
#include <string>
#include <vector>
#include <cstring>
using namespace std;
bool visit[201];
int N;
vector<vector<int>> v;
void dfs(int x) {
	visit[x] = true;
	for (int i = 0; i < N; i++) {
		if (visit[i] == false && v[x][i] == 1) {
			dfs(i);
		}
	}
}
int solution(int n, vector<vector<int>> computers) {
	int answer = 0;
	memset(visit, false, sizeof(visit));
	N = n;
	v = computers;
	for (int i = 0; i < n; i++) {
		if (visit[i] == false) {
			dfs(i);
			answer++;
		}
	}
	return answer;
}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.java
class Solution {
    boolean[] visit;
    int[][] computers;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visit = new boolean[n];
        this.computers = computers;
        for (int i=0; i<n; i++) {
            if (!visit[i]) {
                dfs(i, n);
                answer++;
            }
        }
        return answer;
    }

    private void dfs(int x, int n) {
        visit[x] = true;
        for (int i=0; i<n; i++) {
            if (visit[i] || computers[x][i] == 0) continue;
            dfs(i, n);
        }
    }
}